lib/rules/annotation-no-unknown/index.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const valueParser = require('postcss-value-parser');
+const getDeclarationValue = require('../../utils/getDeclarationValue');
+
+const keywordSets = require('../../reference/keywordSets');
+const optionsMatches = require('../../utils/optionsMatches');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const { isRegExp, isString } = require('../../utils/validateTypes');
+
+const ruleName = 'annotation-no-unknown';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (annotation) => `Unexpected unknown annotation "${annotation}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/annotation-no-unknown',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreAnnotations: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls(checkStatement);
+
+		/**
+		 * @param {import('postcss').Declaration} decl
+		 */
+		function checkStatement(decl) {
+			if (!decl.value.includes('!')) return;
+
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			parsedValue.walk((node) => {
+				if (!isAnnotation(node)) return;
+
+				const value = node.value;
+
+				if (optionsMatches(secondaryOptions, 'ignoreAnnotations', value)) {
+					return;
+				}
+
+				if (keywordSets.annotations.has(value.toLowerCase())) {
+					return;
```

Comment:
This does never get called. It "accidentally" works because a valid `!important` is stripped from the declaration somehow. I'll need to investigate more.